api/envoy/config/filter/fault/v2/fault.proto

```diff
@@ -40,6 +46,9 @@ message FaultDelay {
     // for the specified period. This is required if type is FIXED.
     google.protobuf.Duration fixed_delay = 3
         [(validate.rules).duration.gt = {}, (gogoproto.stdduration) = true];
+
+    // Fault delays are controlled via an HTTP header (if applicable).
+    HeaderDelay header_delay = 5;
```

Comment:
General thought: since header based control negates the need for a percentage, do you think its worthwhile to do this in the way retries are done? I.e. header based ones will take precedence over the ones supplied in config?
This way, the config only needs a boolean flag to state whether or not to accept fault headers..
```
fault:
 delay:
  allowDelayHeaders: true
  fixedDelay: 0s
  percentage: 0
```

We can then state in the comments for percentage field below that the value of percentage will be ignored if there is a fault header in the request (we would also ignore the runtime field I guess?)

Same discussion applies to the rate limit thing as well

Reply:
thinking more about it, ignore the thought about using headers to "override". Unlike retries, there is no default behavior for fault. So, its either

```
fault:
  delay:
    headerDelay: {}
```

or 
```
fault:
  delay:
    fixedDelay: ...
    percentage: ..
```

We dont have to do enforce this, but in the comments, we should mention that percentage will be ignored if headerDelay is set.

Reply:
Personally, I think percentage is still useful for incremental rollout, etc. which is why I prefer to keep percentage generic for all? WDYT?

Reply:
fair point.